tower-balance/examples/demo.rs

```diff
@@ -233,12 +234,25 @@ impl Discover for Disco {
     }
 }
 
+#[derive(Debug)]
+pub struct DiscoError;
```

Comment:
This is all pretty terrible, but I *think* most of these changes will go away *once* `Box<Error>` is applied everywhere.